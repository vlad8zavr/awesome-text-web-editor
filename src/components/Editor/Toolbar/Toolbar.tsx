import React from 'react';
import CustomEditor from '../CustomEditor/CustomEditor';
import { Editor } from 'slate';
import { ReactEditor } from 'slate-react';

interface Props {
    editor: Editor & ReactEditor;
}

const Toolbar = (props: Props) => (
    <div className='toolbar'>
        <span
            className='toolbar__item material-icons css-583gqu'
            onMouseDown={event => CustomEditor.toggleBoldMark(props.editor, event)}
        >format_bold</span>
        <span
            className='toolbar__item material-icons css-583gqu'
            onMouseDown={event => CustomEditor.toggleItalicMark(props.editor, event)}
        >format_italic</span>
        <span
            className='toolbar__item material-icons css-583gqu'
            onMouseDown={event => CustomEditor.toggleCodeBlock(props.editor, event)}
        >code</span>
    </div>
);

export default Toolbar;
