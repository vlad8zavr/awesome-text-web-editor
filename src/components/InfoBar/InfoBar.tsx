import React from 'react';

const InfoBar = () => (
    <div className='info-bar'>
        <div className='info-bar__item'>toggle <strong>code mode</strong>: Ctrl-`</div>
        <div className='info-bar__item'>toggle <strong>bold text</strong>: Ctrl-b</div>
        <div className='info-bar__item'>toggle <strong>italic text</strong>: Ctrl-i</div>
    </div>
);

export default InfoBar;
