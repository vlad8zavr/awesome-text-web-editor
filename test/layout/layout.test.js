import React from 'react';
import { shallow } from 'enzyme';
import InfoBar from '../../src/components/InfoBar/InfoBar';
import Toolbar from '../../src/components/Editor/Toolbar/Toolbar';

describe('layout', () => {

  it('InfoBar should render correctly', () => {
      const component = shallow(<InfoBar />);
      expect(component).toMatchSnapshot();
  });

  it('Toolbar should render correctly', () => {
      const component = shallow(<Toolbar />);
      expect(component).toMatchSnapshot();
  });

});