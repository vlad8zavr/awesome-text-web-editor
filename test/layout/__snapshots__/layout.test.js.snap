// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout InfoBar should render correctly 1`] = `
<div
  className="info-bar"
>
  <div
    className="info-bar__item"
  >
    toggle 
    <strong>
      code mode
    </strong>
    : Ctrl-\`
  </div>
  <div
    className="info-bar__item"
  >
    toggle 
    <strong>
      bold text
    </strong>
    : Ctrl-b
  </div>
  <div
    className="info-bar__item"
  >
    toggle 
    <strong>
      italic text
    </strong>
    : Ctrl-i
  </div>
</div>
`;

exports[`layout Toolbar should render correctly 1`] = `
<div
  className="toolbar"
>
  <span
    className="toolbar__item material-icons"
    onMouseDown={[Function]}
  >
    format_bold
  </span>
  <span
    className="toolbar__item material-icons"
    onMouseDown={[Function]}
  >
    format_italic
  </span>
  <span
    className="toolbar__item material-icons"
    onMouseDown={[Function]}
  >
    code
  </span>
</div>
`;
